format-code

```diff
@@ -23,7 +23,7 @@ cd $(dirname $0)
 for i in $(find . -name 'build*' -prune -o '(' \
 		-name '*.hh' -o -name '*.h' -o -name '*.cc' -o -name '*.c' \
 		')' -print); do
-    if clang-format < $i >| $i.new; then
+    if clang-format-16 < $i >| $i.new; then
```

Comment:
What I'd like to see here is a check that the version is >= 16 since this works with 16 and 17. I think we should just bump the minimum version required whenever a new version changes the output. If I recall, the changes from 15 to 16 were bug fixes. I'd suggest just putting a version check here.

Reply:
The reason why I suggested having a fixed version was that the formatting script never worked for me. I always had to use a modified script with the clang version hard coded.

I have no problem using 17 instead of 16.